src/test/scala/Ch5Spec.scala

```diff
@@ -43,5 +45,37 @@ class Ch5Spec extends FlatSpec with Matchers with CatsLawChecking {
     checkCatsMonoidLaws[Boolean => Int](dataIsEqual[Boolean, Int])
   }
 
+  it should "Problem 4" in {
+    import Problem4._
+
+    // Create instance of semigroup S.
+    // Taken from https://github.com/winitzki/scala-examples/blob/master/chapter05/src/test/scala/example/Chapter05_03_workedExamplesSpec.scala#L138
+    implicit val semigroupIntInstanceNonCommutAssoc = new Semigroup[Int] {
+      override def combine(x: Int, y: Int): Int = nonCommutAssoc(x, y)
+
+      // Budden's function: see F. J. Budden, A Non-Commutative, Associative Operation on the Reals.
+      //   The Mathematical Gazette, Vol. 54, No. 390 (Dec., 1970), pp. 368-372
+      private def nonCommutAssoc(x: Int, y: Int): Int =
+        if (x % 2 == 0) x + y else x - y
+    }
+
+    // Should be able to derive monoid instance for (Int, Double) now
+    checkCatsMonoidLaws[Option[Int]]()
+  }
+
+  it should "Problem 5" in {
+    import Problem5._
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    def dataIsEqual[T](f1: Future[Seq[T]], f2: Future[Seq[T]])(implicit arbT: Arbitrary[T]): Assertion = {
+      Await.result(f1, 1 second) shouldEqual Await.result(f2, 1 second)
+      // for {
+      //   r1 <- f1
+      //   r2 <- f2
+      // } yield r1 shouldEqual r2
```

Comment:
This would run the `shouldEqual` check in the future. Not what we want.

I'm not sure why the "boxed error" occurs. Maybe google it if you feel like.  I think checking laws here is optional because this is just a composition of functors.